src/fetchContributions.js

```diff
@@ -20,104 +66,37 @@ const sortByKey = (key) => (a, b) => `${a[key]}`.localeCompare(`${b[key]}`);
  * resolve CCN references by IDCC from @socialgouv/kali-data
  *
  */
-
-const fetchContributions = async () => {
-  const agreements = await fetch(`${API_URL}/agreements`).then((r) => r.json());
-
-  const answers = await fetch(
-    `${API_URL}/public_answers?select=*&order=updated_at.desc`
-  ).then((r) => r.json());
-
+async function fetchContributions() {
+  /** @type {Contributions.QuestionRaw[]} */
   const questions = await fetch(
-    `${API_URL}/questions?select=*&order=id`
+    `${API_URL}/questions?select=id,value,index,answers:public_answers(id,markdown:value,references:answers_references(title:value,url,dila_id,dila_cid),agreement(name,idcc,parent_id))&order=index`
   ).then((r) => r.json());
 
-  const references = await fetch(`${API_URL}/answers_references`).then((r) =>
-    r.json()
-  );
-
-  const getAnswer = (answerId) => answers.find((q) => q.id === answerId);
-
-  const getAgreement = (agreementId) =>
-    agreements.find((a) => a.id === agreementId);
-
-  const getAgreementIdcc = (agreementId) => {
-    const agreement = getAgreement(agreementId);
-    return agreement && agreement.idcc;
-  };
-
-  // get answer CC details from kali-data
-  const getConventionKali = (answerId) => {
-    const answer = getAnswer(answerId);
-    const agreement = answer.agreement_id && getAgreement(answer.agreement_id);
-    const conventionKali =
-      agreement &&
-      kaliData.find(
-        (convention) =>
-          comparableIdcc(convention.num) === comparableIdcc(agreement.idcc)
-      );
-    return conventionKali;
-  };
-
-  const getReferences = (answerId) =>
-    references
-      .filter((r) => r.answer_id === answerId)
-      .map(({ category, value: title, url }) => {
-        if (category === "agreement") {
-          return {
-            category,
-            title,
-            agreement: getConventionKali(answerId),
-          };
-        }
-        return { title, url, category };
-      })
-      .sort(sortByKey("title"));
-
-  const getConventionsAnswers = (questionId) =>
-    answers
-      .filter((a) => a.question_id === questionId && a.agreement_id !== null)
-      .map((ccAnswer) => ({
-        markdown: ccAnswer.value,
-        idcc: getAgreementIdcc(ccAnswer.agreement_id),
-        references: getReferences(ccAnswer.id),
-      }))
-      .sort((a, b) => parseInt(a.idcc, 10) - parseInt(b.idcc, 10));
-
-  const getGenericAnswer = (questionId) => {
-    const genericAnswer = answers.find(
-      (a) => a.question_id === questionId && a.agreement_id === null
-    );
-    if (!genericAnswer) {
-      return;
+  return /**@type {Contributions.Question[]} */ (questions.flatMap(
+    ({ id, index, value: title, answers }) => {
+      const genericAnswer = getGenericAnswer(answers);
+      if (!genericAnswer) return [];
```

Comment:
`array` when no genericAnswer, `object` otherwise ?

Reply:
returning '[]' in flatMap will remove the entry, so no need to filter question without genericAnswer later